File: appConfig.ts
/**
 * appConfig.ts — Central white-label brand configuration.
 *
 * Every user-facing string and brand value lives here.
 * Swap these out to completely rebrand the app for any family or publisher.
 */
export const APP_CONFIG = {
  // ── App identity (shown in stores, browser tabs, splash screens) ──────────
  appName: 'FamilyHub',
  appShortName: 'FamilyHub',
  appDescription: 'Family management hub for tasks, meals, budget, calendar, and more.',

  // ── Auth screen copy ──────────────────────────────────────────────────────
  loginTagline: 'Welcome back.',
  signupTagline: 'Create your account.',
  onboardingTagline: 'Set up your home.',

  // ── Onboarding ────────────────────────────────────────────────────────────
  createHubLabel: 'Create New Home',
  createHubSubLabel: 'Start fresh for your family',
  joinHubLabel: 'Join Existing Home',
  joinHubSubLabel: 'Join via invite code',
  familyNamePlaceholder: 'e.g., The Smith Family',

  // ── Sidebar / nav ─────────────────────────────────────────────────────────
  defaultUserName: 'Family Member',

  // ── AI features ───────────────────────────────────────────────────────────
  aiCoachName: 'AI Health Coach',

  // ── Internal storage keys (changing these breaks existing installs) ───────
  storageKey: 'lobohub_db',
  activeUserKey: 'lobohub_active_user_id',

  // ── Service worker ────────────────────────────────────────────────────────
  swCacheName: 'lobohub-cache-v1',
  swNotificationTag: 'lobohub',

  // ── Subscription pricing (shown in upgrade prompts and paywalls) ──────────
  trialDays: 14,
  pricing: {
    base: {
      monthly: '$9.99/month',
      annual: '$99/year',
      annualNote: '2 months free',
      label: 'FamilyHub Base',
      description: 'Everything your family needs — tasks, meals, budget, fitness, chores, devotionals, and more.',
    },
    ai: {
      monthly: '$17.99/month',
      annual: '$199/year',
      annualNote: '2 months free',
      label: 'FamilyHub AI',
      description: 'Everything in Base plus AI-powered recipe scraping, fitness plans, smart list categorisation, and more.',
    },
  },

  // ── Theme ─────────────────────────────────────────────────────────────────
  themeColor: '#6366f1',
  backgroundColor: '#fcfcf9',
} as const;

File: lib/config/app_config.dart
/// Central brand configuration for Huddle (customer-facing).
/// Internal / repo name: LoboHub — do not show in UI.
class AppConfig {
  static const String appName = 'Huddle';
  static const String appShortName = 'Huddle';
  static const String appDescription =
      'Family life in one app — tasks, calendar, meals, budget, and more.';
  static const String internalProductName = 'LoboHub';
  static const String appBundleId = 'com.opensolutions.huddle';

  /// Google Play listing (marketing site & in-app “get the app” links).
  static const String googlePlayUrl =
      'https://play.google.com/store/apps/details?id=com.opensolutions.huddle&hl=en_AU';

  // Auth copy
  static const String loginTagline = 'Welcome back.';
  static const String signupTagline = 'Create your account.';
  static const String onboardingTagline = 'Set up your family.';

  // Onboarding (family-first; “home” only where legally/store copy needs it)
  static const String createHubLabel = 'Start a new family';
  static const String createHubSubLabel = 'Invite people anytime with your code';
  static const String joinHubLabel = 'Join a family';
  static const String joinHubSubLabel = 'Use your invite code';
  static const String familyNamePlaceholder = 'e.g., The Smith Family';

  // Auth — create / join family (shared account, not “home” metaphor in headings)
  static const String authOnboardingSubtitle =
      'Create a new family or join one with a code';
  static const String authNameYourFamilyTitle = 'Name your family';
  static const String authNameYourFamilySubtitle =
      'This name is visible to everyone you invite';
  static const String authFamilyNameFieldLabel = 'Family name';
  static const String authCreateFamilyCta = 'Create family';
  static const String authJoinFamilyTitle = 'Join a family';
  static const String authJoinFamilyCta = 'Join family';
  static const String authInvalidInviteCodeFamily =
      'No family found with that code. Check and try again.';

  // Nav
  static const String defaultUserName = 'Family member';
  static const String manageFamilyMembersLabel = 'Family members';
  static const String manageFamilyMembersTitle = 'Family members';
  static const String activityLogLabel = 'Activity log';
  static const String familyPulseLabel = 'Family pulse';

  // Subscription — user-facing plan names (enums stay base / ai / ai_family in code)
  static const String planLabelEssentials = 'Essentials';
  static const String planDescEssentials = 'Core tools for your family';
  static const String planLabelAi = 'AI';
  static const String planDescAi = 'Smart AI features for busy families';
  static const String planLabelAiFamily = 'AI · Family';
  static const String planDescAiFamily =
      'Full AI for up to 2 adults and 2 children under 16';
  static const String planTrialExpiredBlurb =
      'You have Tasks, Lists & Calendar. Upgrade for the full app.';

  // AI
  static const String aiCoachName = 'AI Wellness Assistant';

  // Storage keys
  static const String storageKey = 'lobohub_db';
  static const String activeUserKey = 'lobohub_active_user_id';

  // Subscription
  static const int trialDays = 14;

  // Theme
  static const String themeColorHex = '#6366f1';
  static const String backgroundColorHex = '#fcfcf9';

  // Legal
  static const String privacyPolicyUrl = 'https://huddleapp.com.au/privacy';
  static const String termsOfServiceUrl = 'https://huddleapp.com.au/terms';

  /// Customer support (mailto). Used from Settings → About.
  static const String supportEmail = 'support@huddleapp.com.au';

  // OAuth / deep-link redirects
  static const String oauthRedirectScheme =
      'com.opensolutions.huddle://login-callback';
  static const String passwordResetRedirect =
      'com.opensolutions.huddle://reset-callback';
}
